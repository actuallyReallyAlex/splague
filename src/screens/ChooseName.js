import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Box, Button, FormField, TextInput } from 'grommet'
import {
  chooseName,
  changeScreen,
  transitionScreen
} from '../redux/actions/actions'

export class ChooseName extends Component {
  static propTypes = {
    // From mapStateToProps()
    cure: PropTypes.object.isRequired,
    // From connect()
    dispatch: PropTypes.func.isRequired,
    // From mapStateToProps()
    log: PropTypes.array.isRequired,
    // From mapStateToProps()
    plague: PropTypes.object.isRequired,
    // From mapStateToProps()
    player: PropTypes.object.isRequired,
    // From mapStateToProps()
    ui: PropTypes.object.isRequired,
    // From mapStateToProps()
    world: PropTypes.object.isRequired
  }

  state = {
    error: false,
    name: ''
  }

  playerName = React.createRef()

  handleNameSelection = e => {
    e.preventDefault()
    const { error } = this.state

    if (!error) {
      const { dispatch } = this.props
      const { name } = this.state
      dispatch(chooseName(name))
      dispatch(transitionScreen(true, 'animated fadeOut'))

      setTimeout(() => {
        dispatch(transitionScreen(false, 'animated fadeIn'))
        dispatch(changeScreen('chooseType'))
      }, 1500)
    }
  }

  handleNameChange = e => {
    e.preventDefault()
    const name = e.target.value
    const regex = /\w/gm
    if (regex.test(name)) {
      this.setState(() => ({ name, error: false }))
    } else {
      this.setState(() => ({ error: true }))
    }
  }

  render() {
    const { error, name } = this.state
    const { ui, player } = this.props
    return (
      <Box
        align="center"
        background={player.morality === 'good' ? 'accent-1' : '#252839'}
        className={ui.transitionClasses}
        fill
        justify="center"
      >
        <form>
          <Box align="center">
            <Box>
              <FormField
                error={error && "Name can't be blank. Please try again."}
                label="Name"
              >
                <TextInput
                  onChange={this.handleNameChange}
                  ref={this.playerName}
                />
              </FormField>
            </Box>
            {name && (
              <Box
                className="animated fadeInUp"
                margin={{ top: '130px' }}
                style={{ position: 'absolute' }}
              >
                <Button
                  label="Continue"
                  onClick={this.handleNameSelection}
                  primary
                  type="button"
                />
              </Box>
            )}
          </Box>
        </form>
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseName)
