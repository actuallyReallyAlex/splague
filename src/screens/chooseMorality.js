import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Box, RadioButton } from 'grommet'
import { chooseMorality } from '../redux/actions/actions'

class ChooseMorality extends Component {
  handleMoralitySelection = e => {
    const { dispatch } = this.props
    dispatch(chooseMorality(e.target.name))
  }

  render() {
    const { player } = this.props
    return (
      <Box align="center" fill justify="center">
        <Box direction="row" gap="medium">
          <RadioButton
            checked={player.morality === 'good'}
            label="Good"
            name="good"
            onChange={this.handleMoralitySelection}
          />
          <RadioButton
            checked={player.morality === 'evil'}
            label="Evil"
            name="evil"
            onChange={this.handleMoralitySelection}
          />
        </Box>
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseMorality)
