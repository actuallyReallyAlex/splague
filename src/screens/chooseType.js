import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Box, Button, FormField, Select } from 'grommet'
import { chooseType, changeScreen } from '../redux/actions/actions'
import { plagueTypes, factionTypes } from '../constants'

class ChooseType extends Component {
  state = {
    type: ''
  }

  handleBegin = () => {
    const { dispatch } = this.props
    const { type } = this.state
    dispatch(chooseType(type))
    dispatch(changeScreen('home'))
  }

  handleTypeChange = e => {
    const type = e.value
    this.setState(() => ({ type }))
  }

  render() {
    const { morality } = this.props.player
    const { type } = this.state

    return (
      <Box align="center" fill justify="center">
        <Box>
          <FormField label={morality === 'good' ? 'Faction' : 'Plague Type'}>
            <Select onChange={this.handleTypeChange} options={morality === 'good' ? factionTypes : plagueTypes} value={type} />
          </FormField>
        </Box>
        {type && (
          <Box margin={{ top: 'xlarge' }} style={{ position: 'absolute' }}>
            <Button label="Begin ..." onClick={this.handleBegin} primary />
          </Box>
        )}
      </Box>
    )
  }
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(ChooseType)
