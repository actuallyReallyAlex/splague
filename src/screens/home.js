import React from 'react'
import { connect } from 'react-redux'
import { Box } from 'grommet'
import Sidebar from '../components/Sidebar'

const Home = ({ player, ui }) => {
  return (
    <Box fill>
      <Sidebar />
    </Box>
  )
}

const mapStateToProps = state => {
  return state
}

export default connect(mapStateToProps)(Home)
