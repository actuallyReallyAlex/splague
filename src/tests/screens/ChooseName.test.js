import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ChooseName } from '../../screens/ChooseName'
import developmentState from '../fixtures/developmentState'

describe('<ChooseName />', () => {
  it('Should render snapshot of ChooseName screen.', () => {
    const { player, ui } = developmentState
    const component = renderer
      .create(<ChooseName player={player} ui={ui} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
