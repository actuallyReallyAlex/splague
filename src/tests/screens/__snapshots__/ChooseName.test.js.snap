// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseName /> Should render snapshot of ChooseName screen. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #252839;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(255,255,255,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c5 {
  font-size: 18px;
  line-height: 24px;
}

.c8 {
  box-sizing: border-box;
  font-size: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(255,255,255,0.33);
  border-radius: 4px;
  width: 100%;
  border: none;
}

.c8::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c8::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c8::-moz-placeholder {
  color: #AAAAAA;
}

.c8:-ms-input-placeholder {
  color: #AAAAAA;
}

.c8::-moz-focus-inner {
  border: none;
  outline: none;
}

.c7 {
  position: relative;
  width: 100%;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin-top: 3px;
    margin-bottom: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    border-bottom: solid 1px rgba(255,255,255,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    padding: 0px;
  }
}

<div
  className="animated fadeIn c0"
>
  <form>
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <div
          className="c3"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <div
            className="c4"
          >
            <label
              className="c5"
            >
              Name
            </label>
          </div>
          <div
            className="c6"
          >
            <div
              className="c7"
            >
              <input
                autoComplete="off"
                className="c8"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
      
    </div>
  </form>
</div>
`;
